prow/istio-presubmit.sh

```diff
@@ -90,65 +92,23 @@ else
   # Use the current commit.
   GIT_SHA="$(git rev-parse --verify HEAD)"
 fi
-cd $ROOT
-
-# ./bin/generate-protos.sh || die "Could not generate *.pb.go"
-# if [[ -n $(git status --porcelain) ]]; then
-#     git status
-#     die "Repo has unstaged changes. Re-run ./bin/generate-protos.sh"
-# fi
-
-mkdir -p ~/envoy
-cd ~/envoy
-ISTIO_PROXY_BUCKET=$(sed 's/ = /=/' <<< $( awk '/ISTIO_PROXY_BUCKET =/' $ROOT/WORKSPACE))
-PROXYVERSION=$(sed 's/[^"]*"\([^"]*\)".*/\1/' <<<  $ISTIO_PROXY_BUCKET)
-PROXY=debug-$PROXYVERSION
-wget -qO- https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz | tar xvz
-ln -sf ~/envoy/usr/local/bin/envoy $ROOT/pilot/proxy/envoy/envoy
-cd $ROOT
-
-# go test execution
-# time dep ensure -v
-
-# echo FIXME remove mixer tools exclusion after tests can be run without bazel
-# time go test $(go list ./mixer/... | grep -v /tools/codegen)
-# time go test ./pilot/...
-# time go test ./security/...
-# time go test ./broker/...
-# rm -rf vendor/
 
-if [ "${CI:-}" == 'bootstrap' ]; then
-  # Test harness will checkout code to directory $GOPATH/src/github.com/istio/istio
-  # but we depend on being at path $GOPATH/src/istio.io/istio for imports
-  mv ${GOPATH}/src/istio.io/istio ${GOPATH}/src/github.com
-  ln -sf ${GOPATH}/src/github.com/istio ${GOPATH}/src/istio.io
-  ROOT=${GOPATH}/src/istio.io/istio
-  cd ${GOPATH}/src/istio.io/istio
-fi
+echo 'Pulling down pre-built Envoy'
+fetch_envoy
 
-# Build
+echo 'Building everything'
 ${ROOT}/bin/init.sh
 
-#run_or_die_on_change ./bin/fmt.sh
-
-# bazel test execution
 echo 'Running Unit Tests'
 time bazel test --test_output=all //...
 
-# run linters in advisory mode
+echo 'Running linters in advisory mode'
 SKIP_INIT=1 ${ROOT}/bin/linters.sh
 
-diff=`git diff`
-if [[ -n "$diff" ]]; then
-  echo "Some uncommitted changes are found. Maybe miss committing some generated files? Here's the diff"
-  echo $diff
-  # Do not fail for the changes for now; presubmit bot may share the bazel-genfiles, and that will cause
-  # unrelated failure here randomly. See https://github.com/istio/istio/issues/1689 for the details.
-  # TODO: fix the problem and fail here again.
-  # exit -1
+if [[ -n $(git diff) ]]; then
+  echo "Uncommitted changes found:"
+  die $(git diff)
```

Comment:
Note this is a change from today's semantics. Can we do this yet, or do we need to continue to not die if generated files around? (Is someone else adding a `bazel clean` to this file to keep generated files out of different presubmit runs?)

Reply:
Please remove those sections completely. This is not necessary -- and also, this causes the failures for the build cache things unless #1930 is merged.